modules/infra/submodules/storage/efs_backup_vault.tf

```diff
@@ -60,12 +60,53 @@ resource "aws_iam_role_policy_attachment" "efs_backup_role_attach" {
   policy_arn = data.aws_iam_policy.aws_backup_role_policy[0].arn
 }
 
+resource "terraform_data" "check_backup_role" {
+  count = var.storage.efs.backup_vault.create ? 1 : 0
+
+  provisioner "local-exec" {
+    command     = <<-EOF
+      set -x -o pipefail
+
+      sleep_duration=10
+      iam_role="${aws_iam_role.efs_backup_role[0].arn}"
+
+      check_assume_iam_role(){
+        echo "Checking assume on $iam_role ..."
+        aws sts assume-role --role-arn $iam_role --role-session-name tf-check-role-session || return 1
```

Comment:
The premise is this error 
```
IAM Role arn:aws:iam::501620306788:role/e2e-smoke32385-efs-backup cannot be assumed by AWS Backup
```
I was hoping to closely ensure it could be assumed, but the current session role is not able to assume it so ill change it.